packages/mongodb-mql-model/src/main/kotlin/com/mongodb/jbplugin/mql/Node.kt

```diff
@@ -58,6 +58,8 @@ data class Node<S>(
 
     inline fun <reified C : Component> components(): List<C> = components.filterIsInstance<C>()
 
+    fun componentsWithChildren(): List<HasChildren<*>> = components.filterIsInstance<HasChildren<*>>()
```

Comment:
I think we can safely change this to:

```suggestion
    fun componentsWithChildren(): List<HasChildren<S>> = components.filterIsInstance<HasChildren<S>>()
```

It will have better type checking. Just a quick note: this is not recursive, so it will only get the first level of children. Is this expected?

Reply:
yea this one is only expected to filter the components that have children. the recursive gathering is taken care of by the parser method.